### PR TITLE
Build new donation flow

### PR DESCRIPTION
**Pull request by HSalmon**
[Trello card](https://trello.com/c/utnX8YJR/1002-3-confirm-gift-plan-modal)

screen shot 2015-11-26 at 10 10 46 am

screen shot 2015-11-26 at 10 07 28 am

screen shot 2015-11-26 at 10 15 51 am

### CODE REVIEW

**Comment by HSalmon** [comment]
@mhjb, @rdbartlett, @gdpelican, @robguthrie - ready for review?

**Comment by whit537** [comment]
@HSalmon I don't see where to thank you over on Loomio for approving me to join the Loomio group, but I happened to simultaneously have this PR open in a tab, so ...  thank you! :-)

screen shot 2015-11-25 at 4 45 51 pm

Okay! Sorry to interrupt. :-)

File: lineman/app/components/group_page/trial_card/confirm_gift_plan_modal/confirm_gift_plan_modal.coffee [hunk 1]
```diff
@@ -1,9 +1,13 @@
 angular.module('loomioApp').factory 'ConfirmGiftPlanModal', ->
   templateUrl: 'generated/components/group_page/trial_card/confirm_gift_plan_modal/confirm_gift_plan_modal.html'
   size: 'confirm-gift-plan-modal'
-  controller: ($scope, group) ->
+  controller: ($scope, group, ModalService, ChoosePlanModal, DonationModal) ->
```
Comment (gdpelican) [inline lineman/app/components/group_page/trial_card/confirm_gift_plan_modal/confirm_gift_plan_modal.coffee]: Note to self: at some point we should really move away from needing to inject the ModelService _and_ the modal we want to open.

File: lineman/app/components/group_page/trial_card/donation_modal/donation_modal.coffee [hunk 2]
```diff
@@ -0,0 +1,21 @@
+angular.module('loomioApp').factory 'DonationModal', ->
+  templateUrl: 'generated/components/group_page/trial_card/donation_modal/donation_modal.html'
+  size: 'confirm-gift-plan-modal'
+  controller: ($scope, group, $window, AppConfig, CurrentUser) ->
+    $scope.group = group
+
+    $scope.makeDonation = ->
+      $window.open "#{AppConfig.chargify.donation_url}?#{encodedChargifyParams()}", '_blank'
```
Comment (gdpelican) [inline lineman/app/components/group_page/trial_card/donation_modal/donation_modal.coffee]: Will want to do a check for whether AppConfig.chargify exists here (ie, whether we're on loomio.org or not)

**Comment by gdpelican** [comment]
Everytime I see this code grow, I get a little more nervous about including it in our repo, since it's not really source that anyone wants to download and run on their own.

I'm thinking the payment flow is a really good candidate for our first plugin, to get all of this Chargify-specific cruft out of our whitelabel product.

**Comment by gdpelican** [comment]
Otherwise, looks good though @HSalmon

**Comment by robguthrie** [comment]
@gdpelican I really think that we DO want donation stuff in the open source product. Not so sure about subscription stuff

**Comment by mhjb** [comment]
@HSalmon I'm sorry, InVision Sync is buggy for me at the moment, and the designs are slightly out of date. Can you please add a full stop at the end of the checkbox label, and change 'No thanks' to 'Just take me to my group' ?

File: lineman/app/components/group_page/trial_card/donation_modal/donation_modal.coffee (hunk 2 above)
Comment (HSalmon) [inline lineman/app/components/group_page/trial_card/donation_modal/donation_modal.coffee]: @robguthrie, @gdpelican - responding to your comments - what do we wanna do here? Does this comment still apply?

**Comment by HSalmon** [comment]
@mhjb, done!

screen shot 2015-11-26 at 10 15 38 am

screen shot 2015-11-26 at 11 53 56 am

**Comment by mhjb** [comment]
:)

**Comment by HSalmon** [comment]
@gdpelican, @robguthrie, does this need anything further?

**Comment by gdpelican** [comment]
Yep, will need a little mobile love, looks like:

screen shot 2015-11-27 at 7 39 42 am

**Comment by gdpelican** [comment]
The checkbox in the first modal looks a little not-great, but I reckon we leave it for now, since check box styling isn't trivial. Maybe we'll come up with a directive one day that'll let us drop in a nice-looking, stylable checkbox everywhere we want one.

**Pull request merged**